### PR TITLE
Making profile form more accessible

### PR DESCRIPTION
**Pull request by singingwolfboy**
This pull request does the following things, all related to the profile form:

* Replace `<div class="section-header">` elements with `<h3>` elements
* Add accessible `<label>` elements to the date fields and the select fields
* Ensure that every select field has a unique ID, so that we can use `aria-describedby` attributes to label error messages in a future pull request
* Pull Sass related to the profile form out into its own `profile-form.scss` file, for organizational purposes

To test this pull request, I suggest using [ChromeVox](http://www.chromevox.com/) or [VoiceOver](https://help.apple.com/voiceover/info/guide/10.12/). To activate/deactivate VoiceOver on a Mac, press Command + F5. Different browsers & screen readers work differently; for example, when reading the "Date of birth" fields, both ChromeVox and VoiceOver with Chrome read out "Day, edit text", "Month, edit text", "Year, edit text". However, VoiceOver with Safari reads out "Day, edit text, date of birth", "Month, edit text, date of birth", "Year, edit text, date of birth", which is much clearer for a non-sighted user. (This is related to #1683.)

A few concrete manual test cases:

* For every select field, try clicking the label, and ensure that the browser automatically focuses on the select field.
* Turn on ChromeVox or VoiceOver and tab through the profile form. All fields should announce themselves properly, with the following known exceptions:
  * the "Gender" radio fields are not properly labelled
  * error states, such as not filling in required fields, are not properly accessible

screen shot 2016-11-17 at 11 49 12 am
screen shot 2016-11-17 at 11 48 32 am
screen shot 2016-11-17 at 11 48 37 am

### CODE REVIEW

**Comment by pdpinch** [comment]
Maybe this is moot because this is WIP, but I have no idea what the scope of this is. Does it address a particular issue?

**Comment by noisecapella** [comment]
Can you rebase please?

File: static/js/util/profile_edit_test.js [hunk 1]
```diff
@@ -145,10 +145,11 @@ describe('Profile Editing utility functions', () => {
     });
 
     let rerender = omitDay => {
+      let legend, space;  // eslint-disable-line no-unused-vars
       if (omitDay) {
-        [monthTextField, , yearTextField] = renderDateField(true);
+        [legend, space, monthTextField, , yearTextField] = renderDateField(true);
```
Comment (noisecapella) [inline static/js/util/profile_edit_test.js]: As long as you're here, do you mind refactoring this function to use enzyme, if you feel like it? `renderDateField` is kind of a hacky workaround at the moment

Comment (noisecapella) [inline static/js/util/profile_edit_test.js]: Made #1774 for this

**Comment by noisecapella** [comment]
It looks like the next button on the signup page is now causing a full page refresh. Can you investigate a fix?

**Comment by noisecapella** [comment]
There is a change of appearance in the user page when I edit the personal information. Before:
screenshot from 2016-11-15 15-42-07

After:
screenshot from 2016-11-15 15-41-34

**Comment by singingwolfboy** [comment]
@noisecapella Thanks for catching that. The page refresh was due to me swapping a `<section>` element with a `<form>` element -- looks like introducing true forms is going to be tricker than I expected, and will almost certainly require using the [redux-form](http://redux-form.com/) library, so I've reverted that change. I've also fixed the styling problems with the profile edit form, although there may still be some fiddly form label issues that might need some attention from @roberthouse54 at some point.

**Comment by noisecapella** [comment]
It seems like there's still a regression in the style, though it's smaller than before:

screenshot from 2016-11-15 16-40-54

Note that the text and select items aren't aligned anymore like in the 'before' image above

**Comment by noisecapella** [comment]
The education form is also affected:
screenshot from 2016-11-15 16-42-44

**Comment by noisecapella** [comment]
Please add the 'Needs Review' label back when you are done addressing comments

**Comment by singingwolfboy** [comment]
I spent some time with @roberthouse54 going over the styling in this pull request, and we made some changes to improve it. It's still not pixel-perfect, but it looks much more reasonable now, and nothing seems obviously broken. I've updated the screenshots in the description of this pull request to reflect the way the pull request currently looks.

I'll be out all of next week, so if someone else wants to pick this up while I'm gone, that's fine with me. Otherwise, we can either merge it before I leave for the day, and clean up the little issues in a future pull request; or we can hold off until I get back.

**Comment by noisecapella** [comment]
Looks like the style issues are fixed. I'll proceed with the manual steps in your description

**Comment by noisecapella** [comment]
Looks good, I'll merge this for you :+1:

**Pull request merged**